server/ServerFSM.cpp

```diff
@@ -688,8 +693,34 @@ sc::result MPLobby::react(const LobbyUpdate& msg) {
             if (psd.m_starting_species_name.empty())
                 psd.m_starting_species_name = GetSpeciesManager().RandomPlayableSpeciesName();
         }
+
+        if (psd.m_empire_color != CLR_NONE) {
+            if (!psd_colors.emplace(psd.m_empire_color, entry.first).second) {
+                conflicted_psd_ids.insert(psd_colors.at(psd.m_empire_color));
+                conflicted_psd_ids.insert(entry.first);
+            }
+        }
+
+        if (!psd.m_player_name.empty()) {
```

Comment:
requiring player names (not just empire names) to be unique is problematic because you can't change your player name after joining a server